### ISSUE TITLE
Unify file handling

### ISSUE THREAD

**Issue by xrotwang** [lmaurits/BEASTling#241]
Now that we have py3.5 as baseline, we should unify all file handling to make use of `pathlib`. I.e. filenames should be handled like encoded strings: Convert to `pathlib.Path` as early as possible, and pass around `pathlib.Path` objects internally exclusively.

Also, reading and writing files can then be handled using the `read_text` and `write_text` method, etc.

**Comment by Anaphory** [comment]
Yes! You taught me to love `pathlib.Path`, this sounds like an excellent plan.

**Comment by xrotwang** [comment]
@Anaphory and if you still love it, after the rough years of `pathlib2` and backports for py2, you'll love it happily ever after.
